OrbitCore/TracepointEventBuffer.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TracepointEventBuffer.h"
+
+#include "SamplingProfiler.h"
+
+using orbit_client_protos::TracepointEventInfo;
+
+std::vector<TracepointEventInfo> TracepointEventBuffer::GetTracepointEvents(
+    uint64_t time_begin, uint64_t time_end, ThreadID thread_id /*= kAllThreadsFakeTid*/) {
+  std::vector<TracepointEventInfo> tracepoint_events;
+  for (auto& pair : tracepoint_events_) {
+    const ThreadID tracepoint_thread_id = pair.first;
+    std::map<uint64_t, TracepointEventInfo>& tracepoints = pair.second;
+
+    CHECK(thread_id == SamplingProfiler::kAllThreadsFakeTid || tracepoint_thread_id == thread_id);
+    for (auto it = tracepoints.lower_bound(time_begin); it != tracepoints.end(); ++it) {
+      uint64_t time = it->first;
+      if (time < time_end) {
+        tracepoint_events.push_back(it->second);
+      }
+    }
+  }
+
+  return tracepoint_events;
+}
+
+void TracepointEventBuffer::AddTracepointEvent(uint64_t time, uint64_t tracepoint_hash,
+                                               ThreadID thread_id) {
```

Comment:
Please use `int32_t` instead of `ThreadID`.

Reply:
of course! thank you!